db/migrate/20180619015853_migrate_poll_affected_user_to_about.rb

```diff
@@ -0,0 +1,14 @@
+class MigratePollAffectedUserToAbout < ActiveRecord::Migration[5.1]
+  def change
+    remove_index :polls, :affected_user_id
+    remove_foreign_key :polls, column: :affected_user_id
+
+    rename_column :polls, :affected_user_id, :about_id
+    add_column :polls, :about_type, :string
+
+    Poll.where.not(about_id: nil).update_all(about_type: User.name)
+
+    add_index :polls, [:about_id, :about_type]
+    add_index :polls, [:about_type, :about_id]
```

Comment:
No me di cuenta de esto antes, el "doble índice" lo entiendo, pero creo que prefiero uno solo (según los generadores de paper_trail ```[type, id]```: https://github.com/paper-trail-gem/paper_trail/blob/master/lib/generators/paper_trail/templates/create_versions.rb.erb). Si algún día vemos un explain que indique lo contrario lo agregamos =)

Lo otro que suelo hacer es también agregar los individuales, en esos casos la DB "fallbackea" a esos en una condición rara (como el ```about: nil``` que pusiste en el scope).

Reply:
Rails te agrega los 2 como lo puse cuando haces un polymorphic, por eso suelo agregar los 2. No tiene sentido, ya que por mas que le pegues por uno solo usa el indice doble.
Si metes por ej `select * FROM versions where item_id IS NULL ;` al explain te dice que usa el indice doble.

Reply:
Eso en PostgreSQL ;)

Reply:
Como dice _Ricardito_ "Tenemos el diablo dentro" jaja.

Reply:
Jaja, no se quien es Ricardito pero tiene razón =P